src/probability/martingale.lean

```diff
@@ -353,6 +386,118 @@ end
 
 end submartingale
 
+-- We now prove the converse of the optional stopping theorem
+
+section indicator_stopping_time
+
+/-- Auxilliary definition for `submartingale_of_expected_stopped_value_mono`.  -/
+noncomputable def indicator_stopping_time (i j : ℕ) (s : set α) : α → ℕ :=
+s.indicator i + sᶜ.indicator j
```

Comment:
Is this clearer as `s.piecewise i j`?

Reply:
That was the first change I tried, but then it complains about decidability... and I stopped trying.

Reply:
I'll have a go myself and push something if I get it working

Reply:
I was doing the same thing just now. It works and is just a matter of adding `[decidable_pred (λ k, k ∈ s)]` here and there.

Reply:
I pushed the change. We could even remove the def and state the lemmas about `set.piecewise` applied to constants.

Reply:
I removed the def and changed the lemmas to apply to `piecewise` of constants.